apps/simple_history.py

```diff
@@ -36,33 +36,59 @@ def get_user(request=None, **kwargs):
 
 class HistoricalChangesMixin:
 
-    def diff(self, old_history):
-        if self is not None and old_history is not None:
-            if not isinstance(old_history, type(self)):
-                raise TypeError(
-                    f"unsupported type(s) for diffing: '{type(self)}' and '{type(old_history)}'")
-
-        changes = []
-        changed_fields = []
-        if hasattr(self, 'instance'):
+    def diff(self, old_history, json_fields_only=False):
```

Comment:
Interesting library I will try to integrate it for the JSON section. However, I fear that the only piece of code we might end up saving will be the `HistoricalChangesMixin.build_changes` method.

Reply:
Thinking about it more - we could use ShipmentSerializer to serialize `old` and `new` in `diff_object_fields` to JSON, And then run deepdiff on the two JSON blobs to get the diff. It seems to me that would cover most of the custom diff code, and it would also take care of the issue with trying to map enum properties to enum serializers.